Src/AutoFixtureUnitTest/Kernel/TerminatingWithPathSpecimenBuilderTest.cs

```diff
@@ -295,5 +317,82 @@ public void CreateOnMultipleThreadsConcurrentlyWorks()
             // Teardown
         }
 
+        [Fact]
+        public void ObjectCreationExceptionIsThrownIfCreationFails()
+        {
+            //arrange
+            var request = new object();
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) => throw new Exception()
+            };
+
+            var sut = new TerminatingWithPathSpecimenBuilder(builder);
+            var context = new DelegatingSpecimenContext();
+
+            //act & assert
+            Assert.Throws<ObjectCreationException>(() => sut.Create(request, context));
+        }
+
+        [Fact]
+        public void ExceptionWithCorrectMessageIsThrownIfCreationFails()
+        {
+            //arrange
+            var request = new object();
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) => throw new Exception()
+            };
+
+            var sut = new TerminatingWithPathSpecimenBuilder(builder);
+            var context = new DelegatingSpecimenContext();
+
+            // act & assert
+            var actualException = Assert.Throws<ObjectCreationException>(() => sut.Create(request, context));
+            Assert.Contains("failed with exception", actualException.Message);
+        }
+
+        [Fact]
+        public void InnerBuilderExceptionIsWrappedByCreationException()
+        {
+            // arrange
+            var request = new object();
+            var exception = new Exception();
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) => throw exception
+            };
+
+            var sut = new TerminatingWithPathSpecimenBuilder(builder);
+            var context = new DelegatingSpecimenContext();
+
+            // act & assert
+            var actualEx = Assert.Throws<ObjectCreationException>(() => sut.Create(request, context));
+            Assert.Same(exception, actualEx.InnerException);
+        }
+
+        [Fact]
+        public void NestedExceptionIsWrappedForOneTimeOnly()
+        {
+            // arrange
+            var requests = new[] {new object(), new object(), new object()};
+            var requestQueue = new Queue<object>(requests);
+            var firstRequest = requestQueue.Dequeue();
+
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) => requestQueue.Count > 0
+                    ? c.Resolve(requestQueue.Dequeue())
+                    : throw new InvalidOperationException()
+            };
+
+            var sut = new TerminatingWithPathSpecimenBuilder(builder);
+            // Cause sut to be executed recursively for multiple times.
+            var context = new SpecimenContext(sut);
+
+            //act & assert
```

Comment:
Throughout the AutoFixture codebase, the [Four Phase Test](http://xunitpatterns.com/Four%20Phase%20Test.html) naming convention is used. Particularly, for existing tests, I don't see any reason to switch to [Arrange-Act-Assert](http://wiki.c2.com/?ArrangeActAssert)? 😃